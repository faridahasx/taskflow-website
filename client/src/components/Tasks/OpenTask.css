.open-task-wrapper {
    min-height: calc(var(--vh) - 80px);
    max-height: calc(var(--vh) - 80px);
    overflow: auto;
    padding: 0 5px;
}

.open-task-wrapper .task-dialog-title {
    max-height: 100%;
    padding: 10px 20px;
    margin-top: 5px;
    margin-bottom: 20px;
    background-color: rgba(234, 234, 234, 0.442);
    overflow-wrap: break-word;
    font-size: 19px;
    word-wrap: break-word;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
}


.task-description {
    width: 100%;
    overflow-wrap: break-word;
}

.t-icon {
    min-width: 40px;
    min-height: 40px;
    color: rgb(59, 59, 59);
}

.description-loading {
    min-height: calc(var(--vh) - 200px);
}

@media all and (min-width: 900px) and (min-height: 500px) {
    .open-task-wrapper {
        min-height: calc(((var(--vh) / 100) * 88) - 90px);
        max-height: calc(((var(--vh) / 100) * 88) - 90px);
    }

    .description-loading {
        min-height: calc(((var(--vh) / 100) * 88) - 200px);
    }
}