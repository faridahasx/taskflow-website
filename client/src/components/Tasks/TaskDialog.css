.task-dialog {
    width: 100vw;
    min-width: var(--min-width);
    height: var(--vh);
    min-height: var(--vh);
    overflow: auto;
    cursor: auto;
    background-color: rgb(255, 255, 255);
    padding: 0 20px;
}

.task-dialog::-webkit-scrollbar-track {
    margin: 5px 0;
}

.task-dialog-top {
    padding-top: 2px;
    justify-content: space-between;
}

.task-dialog-title {
    font-weight: 600;
    letter-spacing: 1px;
    width: 100%;
    min-height: 42px;
    border-radius: 7px;
    border: 1px solid rgb(234, 234, 234);
}


@media all and (min-width: 900px) and (min-height: 500px) {
    .task-dialog-top {
        padding-top: 0;
    }

    .task-dialog {
        padding: 15px 20px;
        border-radius: 7px;
        width: 80vw;
        height: calc((var(--vh) / 100) * 88);
        min-height: calc((var(--vh) / 100) * 88);
    }
}