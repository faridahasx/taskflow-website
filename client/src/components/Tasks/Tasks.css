#tasks-wrapper {
    width: 100%;
    background-color: var(--light-grey);
    overflow: hidden;
}

#tasks {
    width: 100vw;
    min-width: var(--min-width);
    min-height: calc(var(--vh) - 130px);
    padding-top: 150px;
    background-color: var(--light-grey);
    align-items: center;
}

#tasks-list {
    width: 100%;
    align-items: center;
}

#loading-tasks {
    width: 100%;
}

.tasks-empty {
    width: 100%;
    height: calc(var(--vh) - 230px);
}

@media all and (max-width: 444px) {
    #tasks {
        padding-top: 170px;
    }
}

@media all and (min-width: 900px) {
    #tasks-wrapper {
        justify-content: center;
        align-items: center;
    }

    #tasks {
        width: 80vw;
        padding-top: 100px;
    }
}