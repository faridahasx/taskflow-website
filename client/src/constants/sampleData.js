import { getDateWithNDaysOfDifference } from "../utils/dateUtils";

const categoriesSample = [
  {
    title: "All",
    tasks: 5,
    _id: "All",
  },
  {
    _id: "Tasks",
    tasks: 5,
    title: "Tasks",
  },
];

let tasksSample = [
  {
    _id: "1",
    title: "Task 1",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-1),
    finishDate: getDateWithNDaysOfDifference(1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "2",
    title: "Task 2",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-2),
    finishDate: getDateWithNDaysOfDifference(-1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "3",
    title: "Task 3",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(2),
    finishDate: getDateWithNDaysOfDifference(3),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },

  {
    _id: "4",
    title: "Task 4",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(2),
    finishDate: getDateWithNDaysOfDifference(4),
    completedAt: getDateWithNDaysOfDifference(3),
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
  {
    _id: "5",
    title: "Login to continue",
    category: "Tasks",
    description: "",
    startDate: getDateWithNDaysOfDifference(-1),
    finishDate: getDateWithNDaysOfDifference(1),
    completedAt: null,
    createdAt: getDateWithNDaysOfDifference(-4),
    updatedAt: getDateWithNDaysOfDifference(-4),
    __v: 0,
  },
];

export { tasksSample, categoriesSample };
