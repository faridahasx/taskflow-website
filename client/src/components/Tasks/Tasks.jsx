// External imports
import { Suspense, lazy, useState } from "react";
import PropTypes from "prop-types";
// Custom hooks
import useNetworkStatus from "../../hooks/useNetworkStatus";
// Components
import OfflineContent from "../OfflineContent";
import LinearTransition from "../Loading/LinearTransition";
import LoadingModal from "../Loading/LoadingModal";
import CircularLoading from "../Loading/CircularLoading";
import Navigation from "../Navigation/Navigation";
import TaskListItem from "./TaskListItem/TaskListItem";
import TasksEmpty from "./TasksEmpty";

// Styles
import "./Tasks.css";

// Lazy-loaded components
const AddTaskForm = lazy(() => import("../../containers/Tasks/AddTaskForm"));
const EditTaskForm = lazy(() => import("../../containers/Tasks/EditTaskForm"));

const Tasks = (props) => {
  // Destructure props
  const {
    tasks,
    loadingRef,
    handleToggleCompleted,
    handleDelete,
    isTransitioning,
    loadMore,
  } = props;
  // Local state
  const [openEditTaskEditor, setOpenEditTaskEditor] = useState({});
  const [openAddTaskEditor, setOpenAddTaskEditor] = useState(false);
  const [openTask, setOpenTask] = useState({});
  const [expandDetailsTaskID, setExpandDetailsTaskID] = useState("");

  const isOnline = useNetworkStatus();

  // Event handlers
  const handleToggleOpenTaskForm = () => {
    setOpenAddTaskEditor(!openAddTaskEditor);
  };
  const handleCloseAddTaskEditor = () => setOpenAddTaskEditor("");
  const handleCloseEditTaskEditor = () => setOpenEditTaskEditor({});

  return (
    <section id="tasks-wrapper" className="column">
      <div id="tasks" className="column">
        <Navigation handleToggleOpenTaskForm={handleToggleOpenTaskForm} />
        {/* Render task list or display appropriate messages */}
        {tasks !== null ? (
          tasks.length > 0 ? (
            <ol id="tasks-list" className="column">
              {tasks.map((t) => (
                // TaskListItem component for each task
                <TaskListItem
                  key={t._id}
                  task={t}
                  openEditTaskEditor={openEditTaskEditor}
                  setOpenEditTaskEditor={setOpenEditTaskEditor}
                  openTask={openTask}
                  setOpenTask={setOpenTask}
                  handleToggleCompleted={handleToggleCompleted}
                  handleDelete={handleDelete}
                  expandDetailsTaskID={expandDetailsTaskID}
                  setExpandDetailsTaskID={setExpandDetailsTaskID}
                />
              ))}
            </ol>
          ) : (
            // Display message when there are no tasks
            <TasksEmpty>No tasks</TasksEmpty>
          )
        ) : (
          <TasksEmpty>
            {!isOnline ? (
              <OfflineContent />
            ) : (
              <CircularLoading data-testid="loading-initial-tasks-indicator" />
            )}
          </TasksEmpty>
        )}

        {isTransitioning === true ? (
          <LinearTransition />
        ) : (
          // Display loading indicator for more tasks
          loadMore && (
            <span
              className="center"
              id="loading-tasks"
              data-testid="loading-more-tasks-indicator"
              ref={loadingRef}
            >
              <CircularLoading />
            </span>
          )
        )}
      </div>
      {/* Conditional rendering of AddTaskForm or EditTaskForm */}
      {openAddTaskEditor ? (
        <Suspense
          fallback={
            <LoadingModal handleClose={() => setOpenAddTaskEditor(false)} />
          }
        >
          <AddTaskForm handleCloseEditor={handleCloseAddTaskEditor} />
        </Suspense>
      ) : (
        openEditTaskEditor._id && (
          <Suspense
            fallback={
              <LoadingModal handleClose={() => setOpenEditTaskEditor({})} />
            }
          >
            <EditTaskForm
              handleCloseEditor={handleCloseEditTaskEditor}
              task={openEditTaskEditor}
            />
          </Suspense>
        )
      )}
    </section>
  );
};

Tasks.propTypes = {
  tasks: PropTypes.oneOfType([PropTypes.array, PropTypes.oneOf([null])]),
  loadingRef: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
  handleToggleCompleted: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  isTransitioning: PropTypes.bool.isRequired,
  loadMore: PropTypes.bool.isRequired,
};

export default Tasks;
