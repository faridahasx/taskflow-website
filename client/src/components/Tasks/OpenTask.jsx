// External imports
import { useEffect } from "react";
import PropTypes from "prop-types";
// Components
import CircularLoading from "../Loading/CircularLoading";
import EditButton from "../IconButtons/EditButton";
import TaskDialog from "./TaskDialog";
// Styles
import "./OpenTask.css";
import "./TaskForm/TextEditor/styles.css";

const OpenTask = (props) => {
  // Destructure props
  const {
    task,
    openEditTaskEditor,
    handleClickEditTask,
    handleCloseOpenedTask,
    loading,
  } = props;

  // useEffect to manage body class for overflow
  useEffect(() => {
    document.body.classList.add("overflow-hidden");
    return () => {
      !openEditTaskEditor._id &&
        document.body.classList.remove("overflow-hidden");
    };
  }, [openEditTaskEditor]);

  return (
    <TaskDialog
      handleBackgroundClick={handleCloseOpenedTask}
      handleCloseClick={handleCloseOpenedTask}
      SecondButton={<EditButton onClick={handleClickEditTask} />}
    >
      <div className="open-task-wrapper">
        <h1 className="task-dialog-title flex">{task.title}</h1>
        <div className="center">
          {loading ? (
            <div className="description-loading center">
              <CircularLoading />
            </div>
          ) : (
            <section
              className="task-description"
              dangerouslySetInnerHTML={{ __html: task.description }}
            />
          )}
        </div>
      </div>
    </TaskDialog>
  );
};

OpenTask.propTypes = {
  task: PropTypes.object.isRequired,
  openEditTaskEditor: PropTypes.object,
  handleClickEditTask: PropTypes.func.isRequired,
  handleCloseOpenedTask: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

export default OpenTask;
