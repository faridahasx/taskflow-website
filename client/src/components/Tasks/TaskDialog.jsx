// External imports
import PropTypes from "prop-types";
// Components
import Modal from "../Modal/Modal";
import CloseButton from "../IconButtons/CloseButton";
// Styles
import "./TaskDialog.css";

const TaskDialog = (props) => {
  // Destructure props
  const { children, handleBackgroundClick, handleCloseClick, SecondButton } =
    props;
  return (
    <Modal handleClose={handleBackgroundClick} className=" task-modal">
      <div className="task-dialog">
        <div className="flex task-dialog-top">
          <CloseButton onClick={handleCloseClick} />
          {SecondButton}
        </div>
        {children}
      </div>
    </Modal>
  );
};

TaskDialog.propTypes = {
  children: PropTypes.any.isRequired,
  SecondButton: PropTypes.oneOfType([PropTypes.elementType, PropTypes.object])
    .isRequired,
  handleBackgroundClick: PropTypes.func.isRequired,
  handleCloseClick: PropTypes.func.isRequired,
};

export default TaskDialog;
