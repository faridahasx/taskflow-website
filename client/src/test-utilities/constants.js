import { MISSING_INPUT_FIELD } from "constants/alertMessages";

// AUTH
export const passwordLabel = "Password:";
export const emailLabel = "Email:";
export const firstnameLabel = "First Name:";
export const lastnameLabel = "Last Name:";

// Server responses
export const successResponse = { status: 200, data: "Success" };

// Redux state
export const reduxStateOnMissingInputField = [
  {
    type: "ALERT",
    payload: MISSING_INPUT_FIELD,
  },
];
