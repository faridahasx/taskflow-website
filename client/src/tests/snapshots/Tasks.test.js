import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Tasks from "../../components/Tasks/Tasks";
import { tasksSample } from "../../constants/sampleData";

const mockFn = jest.fn();

test("Tasks snapshot", () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Tasks
        tasks={tasksSample}
        loadingRef={null}
        handleToggleCompleted={mockFn}
        handleDelete={mockFn}
        isTransitioning={false}
        loadMore={false}
      />
    </BrowserRouter>
  );

  expect(asFragment()).toMatchSnapshot("Tasks");
});
