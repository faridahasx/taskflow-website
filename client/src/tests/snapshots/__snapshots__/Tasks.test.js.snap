// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks snapshot: Tasks 1`] = `
<DocumentFragment>
  <section
    class="column"
    id="tasks-wrapper"
  >
    <div
      class="column"
      id="tasks"
    >
      <nav
        class="flex nav "
        data-testid="nav"
      >
        <div
          class="column"
          id="nav-methods"
        >
          <div
            class="flex"
          >
            <ul
              class="flex"
            >
              <li>
                <span
                  class="nav-item center"
                >
                  <button
                    class="icon "
                    title="Search"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                  </button>
                </span>
              </li>
              <li>
                <span
                  class="nav-item center"
                >
                  <button
                    class="icon "
                    title="Filters"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="FilterListIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
                      />
                    </svg>
                  </button>
                </span>
              </li>
              <li>
                <span
                  class="nav-item center"
                >
                  <button
                    class="icon "
                    title="Sort"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="SwapVertIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M16 17.01V10h-2v7.01h-3L15 21l4-3.99h-3zM9 3 5 6.99h3V14h2V6.99h3L9 3z"
                      />
                    </svg>
                  </button>
                </span>
              </li>
            </ul>
            <button
              class="icon add-task-btn"
              title="Add Task"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                />
              </svg>
            </button>
          </div>
          <ul
            class="flex active-filters-ul"
          >
            <li>
              <button
                class="center activate-filter-btn active-filter"
                title="All tasks"
              >
                All
              </button>
            </li>
            <li>
              <button
                class="center activate-filter-btn inactive-filter"
                title="Current tasks"
              >
                Current
              </button>
            </li>
            <li>
              <button
                class="center activate-filter-btn inactive-filter"
                title="Overdue tasks"
              >
                Overdue
              </button>
            </li>
            <li>
              <button
                class="center activate-filter-btn inactive-filter"
                title="Upcoming tasks"
              >
                Upcoming
              </button>
            </li>
            <li>
              <button
                class="center activate-filter-btn inactive-filter"
                title="Completed tasks"
              >
                Completed
              </button>
            </li>
          </ul>
        </div>
      </nav>
      <ol
        class="column"
        id="tasks-list"
      >
        <li
          class="task-li"
        >
          <div
            class="task-wrapper flex"
          >
            <button
              class="center color blue "
              title="Mark as completed"
            >
              <span
                class="icon center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="column task-middle"
              data-testid="task-clickable"
            >
              <h3
                class="flex"
                title="Task 1"
              >
                Task 1
              </h3>
              <div
                class="details-container column"
              >
                <span
                  class="flex detail"
                  title="Category: Tasks"
                >
                  <span
                    class="flex"
                  >
                    Category:
                  </span>
                  <span
                    class="flex"
                  >
                    Tasks
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Start date: 12/18/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Start date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/18/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Target date: 12/20/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Target date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/20/2023, 12:00 AM
                  </span>
                </span>
              </div>
            </div>
            <div
              class="column task-controls"
            >
              <button
                class="icon expand "
                title="Expand"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </li>
        <li
          class="task-li"
        >
          <div
            class="task-wrapper flex"
          >
            <button
              class="center color red "
              title="Mark as completed"
            >
              <span
                class="icon center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="column task-middle"
              data-testid="task-clickable"
            >
              <h3
                class="flex"
                title="Task 2"
              >
                Task 2
              </h3>
              <div
                class="details-container column"
              >
                <span
                  class="flex detail"
                  title="Category: Tasks"
                >
                  <span
                    class="flex"
                  >
                    Category:
                  </span>
                  <span
                    class="flex"
                  >
                    Tasks
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Start date: 12/17/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Start date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/17/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Target date: 12/18/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Target date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/18/2023, 12:00 AM
                  </span>
                </span>
              </div>
            </div>
            <div
              class="column task-controls"
            >
              <button
                class="icon expand "
                title="Expand"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </li>
        <li
          class="task-li"
        >
          <div
            class="task-wrapper flex"
          >
            <button
              class="center color yellow "
              title="Mark as completed"
            >
              <span
                class="icon center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="column task-middle"
              data-testid="task-clickable"
            >
              <h3
                class="flex"
                title="Task 3"
              >
                Task 3
              </h3>
              <div
                class="details-container column"
              >
                <span
                  class="flex detail"
                  title="Category: Tasks"
                >
                  <span
                    class="flex"
                  >
                    Category:
                  </span>
                  <span
                    class="flex"
                  >
                    Tasks
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Start date: 12/21/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Start date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/21/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Target date: 12/22/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Target date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/22/2023, 12:00 AM
                  </span>
                </span>
              </div>
            </div>
            <div
              class="column task-controls"
            >
              <button
                class="icon expand "
                title="Expand"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </li>
        <li
          class="task-li"
        >
          <div
            class="task-wrapper flex"
          >
            <button
              class="center color green "
              title="Unmark as completed"
            >
              <span
                class="icon center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15yr02a-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="column task-middle"
              data-testid="task-clickable"
            >
              <h3
                class="flex"
                title="Task 4"
              >
                Task 4
              </h3>
              <div
                class="details-container column"
              >
                <span
                  class="flex detail"
                  title="Category: Tasks"
                >
                  <span
                    class="flex"
                  >
                    Category:
                  </span>
                  <span
                    class="flex"
                  >
                    Tasks
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Start date: 12/21/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Start date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/21/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Target date: 12/23/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Target date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/23/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Completed at: 12/22/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Completed at:
                  </span>
                  <span
                    class="flex"
                  >
                    12/22/2023, 12:00 AM
                  </span>
                </span>
              </div>
            </div>
            <div
              class="column task-controls"
            >
              <button
                class="icon expand "
                title="Expand"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </li>
        <li
          class="task-li"
        >
          <div
            class="task-wrapper flex"
          >
            <button
              class="center color blue "
              title="Mark as completed"
            >
              <span
                class="icon center"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="RadioButtonUncheckedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="column task-middle"
              data-testid="task-clickable"
            >
              <h3
                class="flex"
                title="Login to continue"
              >
                Login to continue
              </h3>
              <div
                class="details-container column"
              >
                <span
                  class="flex detail"
                  title="Category: Tasks"
                >
                  <span
                    class="flex"
                  >
                    Category:
                  </span>
                  <span
                    class="flex"
                  >
                    Tasks
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Start date: 12/18/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Start date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/18/2023, 12:00 AM
                  </span>
                </span>
                <span
                  class="flex detail"
                  title="Target date: 12/20/2023, 12:00 AM"
                >
                  <span
                    class="flex"
                  >
                    Target date:
                  </span>
                  <span
                    class="flex"
                  >
                    12/20/2023, 12:00 AM
                  </span>
                </span>
              </div>
            </div>
            <div
              class="column task-controls"
            >
              <button
                class="icon expand "
                title="Expand"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ExpandMoreIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </li>
      </ol>
    </div>
  </section>
</DocumentFragment>
`;
